### ISSUE TITLE
Error when starting a debug session with LPK after elevating permissions for changes to IIS and host file

### ISSUE THREAD

**Issue by irperez** [Azure/dev-spaces#401]
**Describe the bug**
Using AKS and LPK.  Starting a new debug session and after the confirmation to elevate permissions to make changes to disable IIS and make changes to the host file, I get a vague error.
![image](https://user-images.githubusercontent.com/1376080/92679943-cab62500-f2f7-11ea-8a2c-509b3a30ec20.png)

I wish I could see more details about this error. Its vague and I have no clue where to go to figure out what's wrong.

 
**To Reproduce**
Configure LPK using the LPK dialog.  Click the debug button,  hit "OK" on the elevate privledges.
![image](https://user-images.githubusercontent.com/1376080/92680545-14533f80-f2f9-11ea-9438-d33ec0cdad91.png)

 
**Expected behavior**
Browser opens up and normal debugging behavior.
 
**Logs**
Attach logs from the following directory:
For Windows: %TEMP%/Azure Dev Spaces
2020-09-10T03:41:42.6957942Z | SDK | TRACE | Event: DevSpacesManagementClient-EnsureResourceProviderRegistration <json>{"properties":{"resourceProviderNamespace":"Microsoft.DevSpaces","result":"Succeeded"},"metrics":{"duration":1016.0}}</json>\nOperation context: <json>{"clientRequestId":"e4407a42-507b-49e2-8041-5405d984c9a8","correlationRequestId":"9e9eca79-a6a6-4b34-bfd2-fd33643fcb83","requestId":"9e9eca79-a6a6-4b34-bfd2-fd33643fcb83","userSubscriptionId":"HIDDEN","startTime":"2020-09-10T03:41:41.2787891Z","userAgent":"DevSpacesTools/16.0.2007.1401 VisualStudio/16.7.30503.244","requestHttpMethod":null,"requestUri":null,"apiVersion":"3.2","version":"1.0.20200605.3","requestHeaders":{},"loggingProperties":{"applicationName":"SDK","deviceOperatingSystem":"Microsoft Windows 10.0.19041 ","framework":".NET Framework 4.8.4200.0","macAddressHash":"6abecc603326bc21514cd8af735fe793d23e71a2d46c51862ee35c3dcda80bd6","processId":32500,"targetEnvironment":"Production"}}</json>
2020-09-10T03:41:43.1307959Z | SDK | TRACE | Event: DevSpacesManagementClient-ListInSubscription <json>{"properties":{"result":"Succeeded"},"metrics":{"duration":1531.0}}</json>\nOperation context: <json>{"clientRequestId":"e4407a42-507b-49e2-8041-5405d984c9a8","correlationRequestId":"bf58b9b8-e4e3-4ab0-8774-ccb68cd65cbf","requestId":"bf58b9b8-e4e3-4ab0-8774-ccb68cd65cbf","userSubscriptionId":"HIDDEN","startTime":"2020-09-10T03:41:41.2787891Z","userAgent":"DevSpacesTools/16.0.2007.1401 VisualStudio/16.7.30503.244","requestHttpMethod":null,"requestUri":null,"apiVersion":"3.2","version":"1.0.20200605.3","requestHeaders":{},"loggingProperties":{"applicationName":"SDK","deviceOperatingSystem":"Microsoft Windows 10.0.19041 ","framework":".NET Framework 4.8.4200.0","macAddressHash":"6abecc603326bc21514cd8af735fe793d23e71a2d46c51862ee35c3dcda80bd6","processId":32500,"targetEnvironment":"Production"}}</json>

I purposefully removed the userSubscriptionId. FYI.
 
**Environment Details**
Visual Studio
     VS:      v16.7.3
![image](https://user-images.githubusercontent.com/1376080/92680218-5465f280-f2f8-11ea-8aaa-ae3b99220d86.png)

Operating System: Windows 10 Enterprise 19041.450
 
Additional Info:
The AKS environment is using advanced networking and is locked down from a security perspective.  However I have access to the cluster APIs as well via ingress.  Do we need any additional ports open?  Or possibly any other APIs that are not obvious?
![image](https://user-images.githubusercontent.com/1376080/92753997-73946c80-f358-11ea-9685-1912b487e6c0.png)

DevSpaces is off

![image](https://user-images.githubusercontent.com/1376080/92754130-8eff7780-f358-11ea-99ff-7bf1b20083f2.png)

We have a single node pool with 2 servers.

**Comment by Phrow** [comment]
I have the same issue - same VS version, same LPK version. I also tried the "Isolate your local version ..." option, but the issue remains.

I noticed that - while VS is working on establishing the connection, on my AKS, things are happening.
First a new container for the service I want to debug is being created:
![grafik](https://user-images.githubusercontent.com/11168208/92747357-67260900-f384-11ea-8190-950b4fd5d335.png)

Inside that new container, a different image is being pulled and the container initializes successfully.
![grafik](https://user-images.githubusercontent.com/11168208/92747416-77d67f00-f384-11ea-8fed-0ed5ba0715c7.png)

Nevertheless, at that time, VS displays the error message and then restores my previous container, where it first fails because apparently all of a sudden my image can no longer be loaded from the container registry (that's what the error message inside says):
![grafik](https://user-images.githubusercontent.com/11168208/92747598-a5232d00-f384-11ea-8f74-50c8e28820eb.png)
That issue resolves automatically after a while and then I'm back to square one.

**Comment by daniv-msft** [comment]
Thank you @irperez and @Phrow for reporting this issue, and sorry you encountered it.

We would need more logs to investigate this issue, from both Visual Studio and our background CLI.

Getting the logs from our background CLI:
* After reproducing the issue, you'll find the latest logs in:
    * For Windows: `%TEMP%/Local Process with Kubernetes`
    * For OSX/Linux: `$TMPDIR/Local Process with Kubernetes`
* Please take all the logs corresponding to your test (`lpk-library-*`, `lpk-mindarocli-*`, `lpk-endpointmanager-*`).

Getting the logs from our Visual Studio extension:
* Close all Visual Studio windows.
* Set as a system environment variable: `MS_VS_AZUREDEVSPACES_TOOLS_LOGGING_ENABLED`=1
* Start Visual Studio and reproduce the issue.
* Get the logs from `%TEMP%\Microsoft.VisualStudio.Kubernetes.Debugging`

Could you please provide these logs? These logs might contain personally identifiable information, so feel free to remove such information before sending them.
Alternatively, if you don't feel comfortable attaching these logs on this Github issue, feel free to send them to `daniv@microsoft.com` directly.

**Comment by irperez** [comment]
@daniv-msft Thanks for the quick response.  I did see an error in the LPK background CLI logs.

------------------ %TEMP%/Local Process with Kubernetes/lpk-mindarocli ----------------------------
2020-09-10T03:53:16.2098054Z | MindaroCli | TRACE | Event: Command.Start <json>{"properties":{"arguments":"connect --namespace nstanford --service clinicalapi --local-port 5000 --env c:\\users\\iperez\\appdata\\local\\temp\\tmp7731.tmp.env --script c:\\users\\iperez\\appdata\\local\\temp\\tmp7732.tmp --control-port 53602 --ppid 55688 --elevation-requests [{\"targettype\":\"service\",\"targetportinformation\":[{\"name\":\"world wide web publishing service\",\"port\":80,\"processid\":4}],\"requesttype\":\"freeport\"},{\"requesttype\":\"edithostsfile\"}]"},"metrics":null}</json>\nOperation context: <json>{"clientRequestId":null,"correlationRequestId":null,"requestId":null,"userSubscriptionId":null,"startTime":"2020-09-10T03:53:15.9918697Z","userAgent":"LPK/0.1.72.1 VisualStudio/16.7.30503.244","requestHttpMethod":null,"requestUri":null,"version":"1.0.20200825.7","requestHeaders":{},"loggingProperties":{"applicationName":"MindaroCli","deviceOperatingSystem":"Microsoft Windows 10.0.19041","framework":".NET Core 3.1.6","macAddressHash":"6abecc603326bc21514cd8af735fe793d23e71a2d46c51862ee35c3dcda80bd6","processId":29720,"targetEnvironment":"Production","commandId":"6aab5acf-7638-4ba3-8b85-2bf5d8ea0719"}}</json>
**2020-09-10T03:53:16.3043943Z | MindaroCli | ERROR | Unrecognized command or argument 'Wide'\n**
2020-09-10T03:53:16.4100746Z | MindaroCli | WARNG | Logging handled exception: Microsoft.Extensions.CommandLineUtils.CommandParsingException: Serialization Error
2020-09-10T03:53:16.4269562Z | MindaroCli | TRACE | Event: Command.End <json>{"properties":{"arguments":"connect --namespace nstanford --service clinicalapi --local-port 5000 --env c:\\users\\iperez\\appdata\\local\\temp\\tmp7731.tmp.env --script c:\\users\\iperez\\appdata\\local\\temp\\tmp7732.tmp --control-port 53602 --ppid 55688 --elevation-requests [{\"targettype\":\"service\",\"targetportinformation\":[{\"name\":\"world wide web publishing service\",\"port\":80,\"processid\":4}],\"requesttype\":\"freeport\"},{\"requesttype\":\"edithostsfile\"}]","result":"Failed"},"metrics":{"duration":272.0}}</json>



------------  %TEMP%/Local Process with Kubernetes/lpk-library
2020-09-10T03:53:11.2257041Z | Library | TRACE | Dependency: Kubernetes <json>{"target":"ListServicesInNamespaceAsync","success":true,"duration":null,"properties":{}}</json>\nOperation context: <json>{"clientRequestId":null,"correlationRequestId":null,"requestId":null,"userSubscriptionId":null,"startTime":"2020-09-10T03:53:10.4955124Z","userAgent":"LPK/0.1.72.1 VisualStudio/16.7.30503.244","requestHttpMethod":null,"requestUri":null,"version":"1.0.20200825.7","requestHeaders":{},"loggingProperties":{"applicationName":"Library","deviceOperatingSystem":"Microsoft Windows 10.0.19041","framework":".NET Core 3.1.6","macAddressHash":"6abecc603326bc21514cd8af735fe793d23e71a2d46c51862ee35c3dcda80bd6","processId":34240,"targetEnvironment":"Production"}}</json>
2020-09-10T03:53:11.2621158Z | Library | TRACE | Dependency: Kubernetes <json>{"target":"ListServicesInNamespaceAsync","success":true,"duration":null,"properties":{}}</json>
2020-09-10T03:53:11.2682025Z | Library | TRACE | Event: KubernetesRemoteEnvironmentManager-GetReachableServices <json>{"properties":{"result":"Succeeded"},"metrics":null}</json>\nOperation context: <json>{"clientRequestId":null,"correlationRequestId":null,"requestId":null,"userSubscriptionId":null,"startTime":"2020-09-10T03:53:10.4955124Z","userAgent":"LPK/0.1.72.1 VisualStudio/16.7.30503.244","requestHttpMethod":null,"requestUri":null,"version":"1.0.20200825.7","requestHeaders":{},"loggingProperties":{"applicationName":"Library","deviceOperatingSystem":"Microsoft Windows 10.0.19041","framework":".NET Core 3.1.6","macAddressHash":"6abecc603326bc21514cd8af735fe793d23e71a2d46c51862ee35c3dcda80bd6","processId":34240,"targetEnvironment":"Production"}}</json>
2020-09-10T03:53:11.2919837Z | Library | TRACE | Waiting for process 48096
2020-09-10T03:53:11.3048270Z | Library | TRACE | Connecting to EndpointManager
**2020-09-10T03:53:11.3344401Z | Library | ERROR | Logging handled exception: System.Net.Internals.SocketExceptionFactory+ExtendedSocketException: {"Message":"A socket operation encountered a dead network.** C:\\Users\\iperez\\.lpk\\EndpointManager\\EndpointManagerSocket","SocketErrorCode":10050,"ErrorCode":10050,"NativeErrorCode":10050,"StackTrace":"   at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)\r\n   at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)\r\n   at System.Net.Sockets.Socket.UnsafeBeginConnect(EndPoint remoteEP, AsyncCallback callback, Object state, Boolean flowContext)\r\n   at System.Net.Sockets.Socket.BeginConnect(EndPoint remoteEP, AsyncCallback callback, Object state)\r\n   at System.Net.Sockets.Socket.ConnectAsync(EndPoint remoteEP)\r\n   at System.Net.Sockets.SocketTaskExtensions.ConnectAsync(Socket socket, EndPoint remoteEP)\r\n   at Microsoft.DevSpaces.Library.EndpointManagement.EndpointManagerClient.InvokeEndpointManagerAsync[RequestType,ResponseType](RequestType request, CancellationToken cancellationToken, Boolean ensureEndpointManagerRunning)","Data":{},"InnerException":null,"HelpLink":null,"Source":"System.Net.Sockets","HResult":-2147467259}
2020-09-10T03:53:11.3513195Z | Library | WARNG | Logging handled exception: System.InvalidOperationException: {"ClassName":"System.InvalidOperationException","Message":"Could not connect to EndpointManager","Data":null,"InnerException":{"Message":"A socket operation encountered a dead network. C:\\Users\\iperez\\.lpk\\EndpointManager\\EndpointManagerSocket","SocketErrorCode":10050,"ErrorCode":10050,"NativeErrorCode":10050,"StackTrace":"   at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)\r\n   at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)\r\n   at System.Net.Sockets.Socket.UnsafeBeginConnect(EndPoint remoteEP, AsyncCallback callback, Object state, Boolean flowContext)\r\n   at System.Net.Sockets.Socket.BeginConnect(EndPoint remoteEP, AsyncCallback callback, Object state)\r\n   at System.Net.Sockets.Socket.ConnectAsync(EndPoint remoteEP)\r\n   at System.Net.Sockets.SocketTaskExtensions.ConnectAsync(Socket socket, EndPoint remoteEP)\r\n   at Microsoft.DevSpaces.Library.EndpointManagement.EndpointManagerClient.InvokeEndpointManagerAsync[RequestType,ResponseType](RequestType request, CancellationToken cancellationToken, Boolean ensureEndpointManagerRunning)","Data":{},"InnerException":null,"HelpLink":null,"Source":"System.Net.Sockets","HResult":-2147467259},"HelpURL":null,"StackTraceString":"   at Microsoft.DevSpaces.Library.EndpointManagement.EndpointManagerClient.InvokeEndpointManagerAsync[RequestType,ResponseType](RequestType request, CancellationToken cancellationToken, Boolean ensureEndpointManagerRunning)\r\n   at Microsoft.DevSpaces.Library.EndpointManagement.EndpointManagerClient.PingEndpointManagerAsync(CancellationToken cancellationToken)","RemoteStackTraceString":null,"RemoteStackIndex":0,"ExceptionMethod":null,"HResult":-2146233079,"Source":"Microsoft.DevSpaces.Library","WatsonBuckets":null}
2020-09-10T03:53:11.3584259Z | Library | TRACE | Event: ConnectManagementClient-GetElevationRequests <json>{"properties":{"result":"[{\"OccupiedPorts\":[{\"ServiceName\":\"World Wide Web Publishing Service\",\"PortNumber\":80,\"ProcessId\":4}],\"RequestType\":1},{\"RequestType\":0}]","result":"Succeeded"},"metrics":null}</json>



------------------------ %TEMP%\Microsoft.VisualStudio.Kubernetes.Debugging log
{
  "timestamp": "2020-09-10 14-28-05.311",
  "message": "Show the LPK configuration dialog",
  "operationName": "AzureDevSpacesLocalLaunchSettingsViewModel.ShowConfigurationDialog",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-05.363",
  "message": "Initializing the account picker for the cluster selection view model.",
  "operationName": "ConfigurationDialogViewModel.InitializeAccountPickerAsync",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-06.629",
  "message": "Initializing the account picker for the cluster selection view model.",
  "operationName": "ConfigurationDialogViewModel.InitializeAccountPickerAsync",
  "stage": "End",
  "elapsedTime": "00:00:01.2651057",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-08.131",
  "message": "Refreshing the clusters for the selected subscription.",
  "subscriptionId": "HIDDEN",
  "operationName": "ConfigurationDialogCallbacks.RefreshClustersAsync",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-09.076",
  "message": "Refreshing the clusters for the selected subscription.",
  "subscriptionId": "HIDDEN",
  "operationName": "ConfigurationDialogCallbacks.RefreshClustersAsync",
  "stage": "End",
  "elapsedTime": "00:00:00.9439073",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-09.083",
  "message": "Refresh the list of namespaces displayed for the selected cluster",
  "subscriptionId": "HIDDEN",
  "operationName": "ConfigurationDialogCallbacks.RefreshNamespacesAsync",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-16.135",
  "message": "Refresh the list of namespaces displayed for the selected cluster",
  "subscriptionId": "HIDDEN",
  "operationName": "ConfigurationDialogCallbacks.RefreshNamespacesAsync",
  "stage": "End",
  "elapsedTime": "00:00:07.0522943",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-16.144",
  "message": "Refresh the list of services displayed for the selected namespace",
  "subscriptionId": "HIDDEN",
  "operationName": "ConfigurationDialogCallbacks.RefreshServiceNamesAsync",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-16.155",
  "message": "Generate a unique routing isolation header for the current user",
  "operationName": "ConfigurationDialogCallbacks.GetRoutingIsolationHeaderAsync",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-19.272",
  "message": "Generate a unique routing isolation header for the current user",
  "operationName": "ConfigurationDialogCallbacks.GetRoutingIsolationHeaderAsync",
  "stage": "End",
  "elapsedTime": "00:00:03.1167421",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-19.276",
  "message": "Refresh the list of ingress endpoints displayed for the selected namespace",
  "subscriptionId": "HIDDEN",
  "operationName": "ConfigurationDialogCallbacks.RefreshIngressesAsync",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-21.377",
  "message": "Refresh the list of services displayed for the selected namespace",
  "subscriptionId": "HIDDEN",
  "operationName": "ConfigurationDialogCallbacks.RefreshServiceNamesAsync",
  "stage": "End",
  "elapsedTime": "00:00:05.2333491",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-24.069",
  "message": "Refresh the list of ingress endpoints displayed for the selected namespace",
  "subscriptionId": "HIDDEN",
  "operationName": "ConfigurationDialogCallbacks.RefreshIngressesAsync",
  "stage": "End",
  "elapsedTime": "00:00:04.7917707",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-49.352",
  "message": "Show the LPK configuration dialog",
  "operationName": "AzureDevSpacesLocalLaunchSettingsViewModel.ShowConfigurationDialog",
  "stage": "End",
  "elapsedTime": "00:00:44.0409491",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-59.349",
  "message": "Query the debug targets for debug launch of Connect service",
  "devSpacesConfigName": "Local Process with Kubernetes",
  "subscriptionId": "HIDDEN",
  "spaceName": "nstanford",
  "operationName": "AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.QueryDebugTargetsForDebugLaunchAsync",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-59.352",
  "message": "Get the launch profile that is configured to run the service locally",
  "operationName": "AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.GetInnerLaunchProfile",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-59.353",
  "message": "Get the launch profile that is configured to run the service locally",
  "operationName": "AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.GetInnerLaunchProfile",
  "stage": "End",
  "elapsedTime": "00:00:00.0004394",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-59.353",
  "message": "Read the service ports from the target launch profile",
  "operationName": "AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.GetBindingPortsForLaunchProfile",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-59.365",
  "message": "Read the service ports from the target launch profile",
  "operationName": "AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.GetBindingPortsForLaunchProfile",
  "stage": "End",
  "elapsedTime": "00:00:00.0124881",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-59.365",
  "message": "Retrieve the launch targets provider for the specified launch profile",
  "operationName": "AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.GetLaunchTargetsProvider",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-28-59.366",
  "message": "Retrieve the launch targets provider for the specified launch profile",
  "operationName": "AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.GetLaunchTargetsProvider",
  "stage": "End",
  "elapsedTime": "00:00:00.0006704",
  "completionState": "Succeeded"
}
{
  "timestamp": "2020-09-10 14-28-59.370",
  "message": "Begin a connect session",
  "devSpacesConfigName": "Local Process with Kubernetes",
  "projectPath": "C:\\Users\\iperez\\Source\\Repos\\EvitiCMS\\src\\services\\Clinical.API\\Clinical.API\\Clinical.API.csproj",
  "subscriptionId": "HIDDEN",
  "spaceName": "nstanford",
  "operationName": "ConnectionManager.ConnectAsync",
  "stage": "Start"
}
{
  "timestamp": "2020-09-10 14-29-08.650",
  "message": "Begin a connect session",
  "devSpacesConfigName": "Local Process with Kubernetes",
  "projectPath": "C:\\Users\\iperez\\Source\\Repos\\EvitiCMS\\src\\services\\Clinical.API\\Clinical.API\\Clinical.API.csproj",
  "subscriptionId": "HIDDEN",
  "spaceName": "nstanford",
  "exception": {
    "ClassName": "StreamJsonRpc.RemoteInvocationException",
    "Message": "One or more errors occurred.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__118`1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Threading.Tasks.ValueTask`1.get_Result()\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ConnectionManager.ConnectionManager.<>c__DisplayClass26_1.<<ConnectAsync>b__1>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ConnectionManager.ConnectionManager.<>c__DisplayClass26_1.<<ConnectAsync>b__1>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ConnectionManager.ConnectionManager.<>c__DisplayClass26_1.<<ConnectAsync>b__1>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ServiceHub.ConnectServiceBrokerClient.<ExecuteWithServiceAsync>d__4`1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ConnectionManager.ConnectionManager.<>c__DisplayClass26_0.<<ConnectAsync>b__0>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.Logging.PerfLoggerExtensions.<ExecuteOperationAsync>d__3`1.MoveNext()",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": "8\nMoveNext\nStreamJsonRpc, Version=2.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a\nStreamJsonRpc.JsonRpc+<InvokeCoreAsync>d__118`1\nVoid MoveNext()",
    "HResult": -2146233088,
    "Source": "StreamJsonRpc",
    "WatsonBuckets": null,
    "ErrorCode": -32000
  },
  "operationName": "ConnectionManager.ConnectAsync",
  "stage": "End",
  "elapsedTime": "00:00:09.2759163",
  "completionState": "Error"
}
{
  "timestamp": "2020-09-10 14-29-08.650",
  "message": "Query the debug targets for debug launch of Connect service",
  "devSpacesConfigName": "Local Process with Kubernetes",
  "subscriptionId": "HIDDEN",
  "spaceName": "nstanford",
  "exception": {
    "ClassName": "StreamJsonRpc.RemoteInvocationException",
    "Message": "One or more errors occurred.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__118`1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Threading.Tasks.ValueTask`1.get_Result()\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ConnectionManager.ConnectionManager.<>c__DisplayClass26_1.<<ConnectAsync>b__1>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ConnectionManager.ConnectionManager.<>c__DisplayClass26_1.<<ConnectAsync>b__1>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ConnectionManager.ConnectionManager.<>c__DisplayClass26_1.<<ConnectAsync>b__1>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ServiceHub.ConnectServiceBrokerClient.<ExecuteWithServiceAsync>d__4`1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.ConnectionManager.ConnectionManager.<>c__DisplayClass26_0.<<ConnectAsync>b__0>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.Logging.PerfLoggerExtensions.<ExecuteOperationAsync>d__3`1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.LaunchProfiles.AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.<>c__DisplayClass14_0.<<QueryDebugTargetsForDebugLaunchAsync>b__1>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.VisualStudio.Kubernetes.Debugging.Core.Services.Logging.PerfLoggerExtensions.<ExecuteOperationAsync>d__3`1.MoveNext()",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": "8\nMoveNext\nStreamJsonRpc, Version=2.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a\nStreamJsonRpc.JsonRpc+<InvokeCoreAsync>d__118`1\nVoid MoveNext()",
    "HResult": -2146233088,
    "Source": "StreamJsonRpc",
    "WatsonBuckets": null,
    "ErrorCode": -32000
  },
  "operationName": "AzureDevSpacesLocalDebugProfileLaunchTargetsProvider.QueryDebugTargetsForDebugLaunchAsync",
  "stage": "End",
  "elapsedTime": "00:00:09.3010128",
  "completionState": "Error"
}

**Comment by daniv-msft** [comment]
Thank you @irperez for the extra logs. The bug seems to be due to us poorly handling the spaces in the Windows service to stop (`world wide web publishing service` here).
I'm logging a bug on our side to improve this, and I'll reply to this issue once we have the fix.

In the meantime, feel free to manually stop the service by opening the "Services" app on Windows and stopping it from there before starting the Local Process with Kubernetes session.

**Comment by irperez** [comment]
That did the trick!

Thank you @daniv-msft for responding so timely!  This feature is great!

**Comment by daniv-msft** [comment]
Thank you @irperez! That's great to hear!

@Phrow > Could you please confirm that your issue was also related to the World Wide Web Publishing service?
If yes, could you please try to stop it manually through the `Services` app to unblock you while we work on the fix?

**Comment by Phrow** [comment]
Hi @daniv-msft, stopping the service manually beforehand resolved that very issue for me, too. Thanks for the quick response!

Now that it's actually starting the debugging session, I ran into another issue in that my config maps that are mounted as config files/volumes in my services on the aks cluster are not taken into account - and thus my configs are incomplete and the app fails. I could resolve that by copying the missing settings to my local config, but maybe config maps would be interesting to connect, too (besides volumes etc.) - if that's possible.

**Comment by daniv-msft** [comment]
Hi @Phrow,
Thank you for your reply and confirming that stopping the service allowed you to bypass this bug! We're working on it, and hope to have a fix soon.

Regarding your other question: if your config maps are mounted as volumes, we will be able to download them automatically when you start Local Process with Kubernetes.

To do this, you need to create a `KubernetesLocalProcessConfig.yaml` file at the root of your project. You can learn more here: https://docs.microsoft.com/visualstudio/containers/configure-local-process-with-kubernetes?view=vs-2019

In your case, the file's content should look like:
```
version: 0.1
env:
  - name: MY_CONFIG_MAP_PATH
    value: $(volumeMounts:my-config-map)
```
where `my-config-map` is the name of the volume mount, not necessarily the name of the actual config map.

**Comment by daniv-msft** [comment]
@irperez, @Phrow The fix related to us failing when trying to stop "World Wide Web publishing service" has been released. You shouldn't need to stop this service manually anymore if you install the latest Visual Studio extension (now called "Bridge to Kubernetes").

@Phrow Please don't hesitate to let us know if you need any other information to make the config map scenario work.
